src/models/AbstractStyledComponent.js

```diff
@@ -12,10 +12,19 @@ export default class AbstractStyledComponent extends Component {
     generatedClassName?: string,
     generatedStyles?: any
   }
-  unsubscribe: () => void
+
+  unsubscribeId: number = -1
+
+  unsubscribeFromContext() {
```

Comment:
This component is basically to be deleted, since we already have the "BaseStyledComponent":

https://github.com/styled-components/styled-components/pull/1048/files#diff-8a1d0988ee5ae37265b4647fa41def2aR41

I'm not sure whether it's still worth keeping around, even if this code is actually going to be shared with the native implementation as well. The extra class actually seemed to slow instance instantiation down for me in some of my older perf flamegraphs, so I think we should aim for deleting it?